nncf/quantization_metric.py

```diff
@@ -0,0 +1,211 @@
+        self.stat[self.TOTAL_ACTIVATIONS_RATIO] = self.stat[self.ACTIVATIONS_RATIO]
+
+
+    def get_metric_table(self):
+        table = Texttable()
+        data = [self.header]
+        for p in self.params:
+            drow = {h: 0 for h in self.header}
+            for column_name in self.header[1:3]:
+                drow[column_name] = (self.stat[column_name][p] / self.num_all_quantizations) * 100
+            for column_name in self.header[3:]:
+                drow[column_name] = (self.stat[column_name][p] / self.num_all_quantizations) * 100
+            drow[self.PARAMS] = p
+            row = [drow[h] for h in self.header]
+            data.append(row)
+        table.add_rows(data)
+        return table
+
+class MemoryСostMetric(BaseMetric):
+    PARAMS = 'params'
+    NAME = 'MemoryCost'
+
+    EXPECTED_IMPROVEMENT_COST_MEMORY = 'expected improvement cost memory'
+    MAX_TENSOR_ACTIVATIONS_SHAPE = 'max_tensor_activations_shape'
+    COST_MEMORY_FP_WEIGHTS = 'cost memory fp weights'
+    COST_MEMORY_COMPRESSED_WEIGHTS = 'cost memory compressed weights'
+
+    def __init__(self, compressed_model: NNCFNetwork, weights_quantizers, non_weights_quantizers):
+        super().__init__()
+        self._compressed_model = compressed_model
+        self.weights_quantizers, self.non_weights_quantizers = weights_quantizers, non_weights_quantizers
+        self.header = [self.EXPECTED_IMPROVEMENT_COST_MEMORY]
+        self.stat = {}
+
+    def collect(self):
+        self.stat[self.COST_MEMORY_FP_WEIGHTS] = 0
+        self.stat[self.COST_MEMORY_COMPRESSED_WEIGHTS] = 0
+        fp_num_bits = 32 # only cpu ?
+        nncf_modules = self._compressed_model.get_nncf_modules()
+
+        for nncf_module in nncf_modules.values():
+            count_el = np.prod(nncf_module.weight.shape)
+            self.stat[self.COST_MEMORY_FP_WEIGHTS] += count_el * fp_num_bits
+            if len(nncf_module.pre_ops) > 0:
+                # pylint: disable=protected-access
+                num_bits = nncf_module.pre_ops._modules['0'].op.num_bits
+                self.stat[self.COST_MEMORY_COMPRESSED_WEIGHTS] += count_el * num_bits
+            else:
+                self.stat[self.COST_MEMORY_COMPRESSED_WEIGHTS] += count_el * fp_num_bits
+        self.stat[self.EXPECTED_IMPROVEMENT_COST_MEMORY] = self.stat[self.COST_MEMORY_FP_WEIGHTS] /\
+             self.stat[self.COST_MEMORY_COMPRESSED_WEIGHTS]
+
+        self.stat[self.MAX_TENSOR_ACTIVATIONS_SHAPE] = max(self._compressed_model.input_infos,\
+             key=lambda k: np.prod(k.shape)).shape
+
+    def get_metric_table(self):
+        table = Texttable()
+        data = [self.header]
+        drow = {h: 0 for h in self.header}
+        for column_name in self.header:
+            drow[column_name] = self.stat[column_name]
+        row = [drow[h] for h in self.header]
+        data.append(row)
+        table.add_rows(data)
+        return table
+
+
+class ShareBlueEdgesCompressedGraph(BaseMetric):
```

Comment:
Ok, you shouldn't call these edges "blue" anywhere. They are counted not because they are blue, but because they represent a quantized data path.